File: 84.py
def backtrack(nums, path, used, res):
    if len(path) == len(nums):
        res.append(path[:])
        return

    for i in range(len(nums)):
        if used[i]:
            continue
        used[i] = True
        path.append(nums[i])
        backtrack(nums, path, used, res)
        path.pop()
        used[i] = False

nums = [1, 2]
res = []
used = [False] * len(nums)

backtrack(nums, [], used, res)
for r in res:
    print(r)

File: 84.js
function backtrack(nums, path, used, res) {
    if (path.length === nums.length) {
        res.push([...path]);
        return;
    }

    for (let i = 0; i < nums.length; i++) {
        if (used[i]) continue;

        used[i] = true;
        path.push(nums[i]);
        backtrack(nums, path, used, res);
        path.pop();
        used[i] = false;
    }
}

const nums = [1, 2];
const used = Array(nums.length).fill(false);
const res = [];

backtrack(nums, [], used, res);
console.log(res);
